### ISSUE TITLE
requirements-docs.txt installs dask-core

### ISSUE THREAD

**Issue by eracle** [dask/dask#6266]
**What happened**:
Following [the tutorial](https://docs.dask.org/en/latest/develop.html#contributing-to-documentation) on how to build docs locally, I find out that it actually installs dask-core as dependency, which is fine, since it makes the build working, but it then goes in disagreement with the following statement: 

> You can now make edits to rst files and run make html again to update the affected pages.

Since the imported version of dask, and its changed docstrings, are not imported nor build, so new updated docstrings will not be included on the built html.

  
**What you expected to happen**:
dask-core shouldn't be installed or at least we should mention this corner case on [the tutorial](https://docs.dask.org/en/latest/develop.html#contributing-to-documentation) 

**Minimal Complete Verifiable Example**:
```bash
cd docs
conda env create -f requirements-docs.txt -n docs-test
conda activate docs-test
conda list
# packages in environment at /home/eracle/anaconda3/envs/docs-test:
#
# Name                    Version                   Build  Channel
[...]
cytoolz                   0.10.1           py38h516909a_0    conda-forge
dask-core                 2.17.2                     py_0    conda-forge
dask-sphinx-theme         1.3.1                      py_0    conda-forge
distributed               2.17.0           py38h32f6830_0    conda-forge
[...]
```
change any docstring on the codebase
```bash
make html
```
open an html file who is including the issued docstring;
verify it didn't changed.

**Anything else we need to know?**:
If this is ok, I can handle this and create a PR adding few lines on the tutorial.

**Environment**:

- Dask version: 2.16.0+32.g24ad0f0e.dirty
- Python version: 3.8.2
- Operating System: Ubuntu 18.04.4 LTS
- Install method (conda, pip, source): conda

**Comment by eracle** [comment]
btw I solved removing dask-core and installing local version with:

`python setup.py install --record files.txt`

**Comment by jsignell** [comment]
This is probably worth some discussion. I am not sure when we build docs relative to when we release, but I imagine it is kind of nice fail-safe to be able to edit the docs and rebuild them knowing that they will only ever relate to published versions of dask. 

Maybe it'd be more conservative to change the docs-building documentation to recommend doing `pip intall -e ..` after installing dependencies.

**Comment by eracle** [comment]
I agree on being conservative and add a few lines on the docs-building documentation.
Any case I don't think it was a planned behaviour including dask-core as dependency of the docs building dependencies, also because is added through inclusion of distributed (you can check [here](https://github.com/dask/distributed/blob/master/requirements.txt#L4)).
I think we can stick on the easiest and conservative solution, even though I think best solution is to build local changes when the build is made locally. 
We should double check if this is not a problem when new releases are made toward pipy + readthedocs.org (which is where I suppose the dask.com domain points).

**Comment by jsignell** [comment]
Another issue just came up detailing some odd behavior. Probably worth talking about this with @dask/maintenance.

**Comment by eracle** [comment]
Hey @jsignell , thanks for the mention.
You can check out [here](https://readthedocs.org/projects/dask/builds/11184087/) dask.com points to docs.dask.com, which is built by readthedocs everytime a new commit is done on master branch, so everytime we merge.
As you pointed out, merged docstring who weren't released yet are missing on-line.  
I am PR a fix, here's a pic of a local working build:
![to_sql](https://user-images.githubusercontent.com/3613943/83927832-616dc580-a78e-11ea-85df-c33c9d682223.png)

**Comment by jsignell** [comment]
Closed by #6295
